ballerina/tests/from_json_string_test.bal

```diff
@@ -0,0 +1,1072 @@
+    test:assertEquals(x.publisher.year, 1960);
+    test:assertEquals(x.publisher["location"], "Philadelphia");
+    test:assertEquals(x["price"], 10.5);
+    test:assertEquals(x.author["local"], false);
+}
+
+@test:Config
+isolated function testFromJsonStringWithType4() returns error? {
+    string str = string `{
+        "name": "School Twelve",
+        "city": 23,
+        "number": 12,
+        "section": 2,
+        "flag": true,
+        "tp": 12345
+    }`;
+
+    School x = check fromJsonStringWithType(str);
+    test:assertEquals(x.length(), 6);
+    test:assertEquals(x.name, "School Twelve");
+    test:assertEquals(x.number, 12);
+    test:assertEquals(x.flag, true);
+    test:assertEquals(x["section"], 2);
+    test:assertEquals(x["tp"], 12345);
+}
+
+@test:Config
+isolated function testFromJsonStringWithType5() returns error? {
+    string str = string `{
+        "intValue": 10,
+        "floatValue": 10.5,
+        "stringValue": "test",
+        "decimalValue": 10.50,
+        "doNotParse": "abc"
+    }`;
+
+    TestRecord x = check fromJsonStringWithType(str);
+    test:assertEquals(x.length(), 5);
+    test:assertEquals(x.intValue, 10);
+    test:assertEquals(x.floatValue, 10.5f);
+    test:assertEquals(x.stringValue, "test");
+    test:assertEquals(x.decimalValue, 10.50d);
+    test:assertEquals(x["doNotParse"], "abc");
+}
+
+@test:Config
+isolated function testFromJsonStringWithType6() returns error? {
+    string str = string `{
+        "id": 1,
+        "name": "Class A",
+        "student": {
+            "id": 2,
+            "name": "John Doe",
+            "school": {
+                "name": "ABC School",
+                "address": {
+                    "street": "Main St",
+                    "city": "New York"
+                }
+            }
+        },
+        "teacher": {
+            "id": 3,
+            "name": "Jane Smith"
+        },
+        "monitor": null
+    }`;
+
+    Class x = check fromJsonStringWithType(str);
+    test:assertEquals(x.length(), 5);
+    test:assertEquals(x.id, 1);
+    test:assertEquals(x.name, "Class A");
+    test:assertEquals(x.student.length(), 3);
+    test:assertEquals(x.student.id, 2);
+    test:assertEquals(x.student.name, "John Doe");
+    test:assertEquals(x.student.school.length(), 2);
+    test:assertEquals(x.student.school.name, "ABC School");
+    test:assertEquals(x.student.school.address.length(), 2);
+    test:assertEquals(x.student.school.address.street, "Main St");
+    test:assertEquals(x.student.school.address.city, "New York");
+    test:assertEquals(x.teacher.length(), 2);
+    test:assertEquals(x.teacher.id, 3);
+    test:assertEquals(x.teacher.name, "Jane Smith");
+    test:assertEquals(x.monitor, null);
+}
+
+@test:Config
+isolated function testFromJsonStringWithType7() returns error? {
+    string nestedJsonStr = string `{
+        "intValue": 5,
+        "floatValue": 2.5,
+        "stringValue": "nested",
```

Comment:
Shall we add a test with an array in the string?

Reply:
You are mentioning this https://github.com/prakanth97/module-ballerina-data.jsondata/blob/4503830ad162443a2838d3f4ba1d3353f1b47bd4/ballerina/tests/from_json_string_test.bal#L848 right? 
We have similar tests.